modules/rostests/apitests/atl/AtlObjMap.cpp

```diff
@@ -0,0 +1,86 @@
+/*
+ * PROJECT:     ReactOS api tests
+ * LICENSE:     LGPL-2.1-or-later (https://spdx.org/licenses/LGPL-2.1-or-later)
+ * PURPOSE:     Test for OBJECT_ENTRY_AUTO
+ * COPYRIGHT:   Copyright 2023 Mark Jansen <mark.jansen@reactos.org>
+ */
+
+#ifdef HAVE_APITEST
+    #include <apitest.h>
+#else
+    #include "atltest.h"
+#endif
+
+#include <atlbase.h>
+#include <atlcom.h>
+
+class CAtlObjMapModule : public ATL::CAtlExeModuleT<CAtlObjMapModule>
+{
+
+} _Module;
+
+CLSID CLSID_ObjMapTestObject = {0xeae5616c, 0x1e7f, 0x4a00, {0x92, 0x27, 0x1a, 0xad, 0xad, 0x66, 0xbb, 0x85}};
+
+static LONG g_Created = 0;
+static LONG g_Destroyed = 0;
+
+
+START_TEST(AtlObjMap)
+{
+    HRESULT hr = CoInitialize(NULL);
+    ok_hex(hr, S_OK);
+
+    {
+        CComPtr<IUnknown> spTestObject;
+        hr = CoCreateInstance(CLSID_ObjMapTestObject, NULL, CLSCTX_INPROC_SERVER, IID_IUnknown, (LPVOID*)&spTestObject);
+        ok_hex(hr, REGDB_E_CLASSNOTREG);
+    }
```

Comment:
Is this (and the other one too) in a separate code block in order to exercise the ctor/dtor of the spTestObject?

Reply:
yes